federation/pkg/kubefed/init/init_test.go

```diff
@@ -77,6 +77,7 @@ func TestInitFederation(t *testing.T) {
 		image              string
 		etcdPVCapacity     string
 		expectedErr        string
+		dryRun             string
```

Comment:
the reason i kept string is to indicate the type of test case
"" = no dry run test
"valid-run" test for dry run when run passes
"invalid-run" test for dry run when dry run fails for something (how ever did not get a clear way of implementing this test using the current fake factory, so kept as TODO)